lib/trie/trie.go

```diff
@@ -1147,8 +1364,32 @@ func handleDeletion(branch *Node, key []byte) (newNode *Node, branchChildMerged
 			}
 		}
 
-		return newBranch, branchChildMerged
+		return newBranch, branchChildMerged, nil
+	}
+}
+
+// ensureMerkleValueIsCalculated is used before calling PopulateMerkleValues
+// to ensure the parent node and all its descendant nodes have their Merkle
+// value computed and ready to be used. This has a close to zero performance
+// impact if the parent node Merkle value is already computed.
+func (t *Trie) ensureMerkleValueIsCalculated(parent *Node) (err error) {
+	if parent == nil {
+		return nil
 	}
+
+	if parent == t.root {
+		_, err = parent.CalculateRootMerkleValue()
+		if err != nil {
+			return fmt.Errorf("calculating Merkle value of root node: %w", err)
+		}
+	} else {
+		_, err = parent.CalculateMerkleValue()
+		if err != nil {
+			return fmt.Errorf("calculating Merkle value of node: %w", err)
+		}
+	}
+
+	return nil
```

Comment:
nit change to early return

Reply:
Meh I'll keep as it is, this only saves 1 line and it's still short blocks of code.